[{"authorTime":"2018-01-20 05:49:27","codes":[{"authorDate":"2018-01-20 05:49:27","commitOrder":1,"curCode":"    public void rebootVM(final Connection conn, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.cleanRebootAsync(conn);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Reboot VM catch HandleInvalid and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            s_logger.debug(\"Unable to Clean Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString() + \", try hard reboot\");\n            try {\n                vm.hardReboot(conn);\n            } catch (final Exception e1) {\n                final String msg = \"Unable to hard Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n                s_logger.warn(msg, e1);\n                throw new CloudRuntimeException(msg);\n            }\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","date":"2018-01-20 05:49:27","endLine":4465,"groupId":"24018","id":1,"instanceNumber":1,"isCurCommit":0,"methodName":"rebootVM","params":"(finalConnectionconn@finalVMvm@finalStringvmName)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/4b/7080aa5ab2c6da5e2ecf0c8477979016cdb077.src","preCode":"    public void rebootVM(final Connection conn, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.cleanRebootAsync(conn);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Reboot VM catch HandleInvalid and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            s_logger.debug(\"Unable to Clean Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString() + \", try hard reboot\");\n            try {\n                vm.hardReboot(conn);\n            } catch (final Exception e1) {\n                final String msg = \"Unable to hard Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n                s_logger.warn(msg, e1);\n                throw new CloudRuntimeException(msg);\n            }\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","realPath":"plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":4432,"status":"B"},{"authorDate":"2018-01-20 05:49:27","commitOrder":1,"curCode":"    public void startVM(final Connection conn, final Host host, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.startOnAsync(conn, host, false, true);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch HandleInvalid and VM is not in RUNNING state\");\n            } catch (final TimeoutException e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch BadAsyncResult and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            final String msg = \"Unable to start VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n            s_logger.warn(msg, e);\n            throw new CloudRuntimeException(msg);\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","date":"2018-01-20 05:49:27","endLine":5072,"groupId":"4521","id":2,"instanceNumber":2,"isCurCommit":0,"methodName":"startVM","params":"(finalConnectionconn@finalHosthost@finalVMvm@finalStringvmName)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/4b/7080aa5ab2c6da5e2ecf0c8477979016cdb077.src","preCode":"    public void startVM(final Connection conn, final Host host, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.startOnAsync(conn, host, false, true);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch HandleInvalid and VM is not in RUNNING state\");\n            } catch (final TimeoutException e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch BadAsyncResult and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            final String msg = \"Unable to start VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n            s_logger.warn(msg, e);\n            throw new CloudRuntimeException(msg);\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","realPath":"plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":5036,"status":"B"}],"commitId":"893a88d225276e45f12f9490e6af2c94a81c2965","commitMessage":"@@@CLOUDSTACK-10105: Use maven standard project structure in all projects (#2283)\n\nRemove maven standard module (which only a few were using) and get ride of maven customization for the projects structure.\n\n- moved all directories to src/main/java.  src/main/resources.  src/main/scripts.  src/test/java.  src/test/resources\n- grep scan to search for src/com and src/org left over\n- grep for <project>/scripts to fix pom.xml configuration\n- remove custom <build> configuration in pom.xml\n\nSigned-off-by: Marc-Aur?le Brothier <m@brothier.org>","date":"2018-01-20 05:49:27","modifiedFileCount":"1","status":"B","submitter":"Marc-Aur?le Brothier"},{"authorTime":"2018-04-16 15:07:16","codes":[{"authorDate":"2018-01-20 05:49:27","commitOrder":2,"curCode":"    public void rebootVM(final Connection conn, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.cleanRebootAsync(conn);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Reboot VM catch HandleInvalid and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            s_logger.debug(\"Unable to Clean Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString() + \", try hard reboot\");\n            try {\n                vm.hardReboot(conn);\n            } catch (final Exception e1) {\n                final String msg = \"Unable to hard Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n                s_logger.warn(msg, e1);\n                throw new CloudRuntimeException(msg);\n            }\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","date":"2018-01-20 05:49:27","endLine":4465,"groupId":"102935","id":3,"instanceNumber":1,"isCurCommit":0,"methodName":"rebootVM","params":"(finalConnectionconn@finalVMvm@finalStringvmName)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/4b/7080aa5ab2c6da5e2ecf0c8477979016cdb077.src","preCode":"    public void rebootVM(final Connection conn, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.cleanRebootAsync(conn);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Reboot VM catch HandleInvalid and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            s_logger.debug(\"Unable to Clean Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString() + \", try hard reboot\");\n            try {\n                vm.hardReboot(conn);\n            } catch (final Exception e1) {\n                final String msg = \"Unable to hard Reboot VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n                s_logger.warn(msg, e1);\n                throw new CloudRuntimeException(msg);\n            }\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","realPath":"plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":4432,"status":"N"},{"authorDate":"2018-04-16 15:07:16","commitOrder":2,"curCode":"    public void startVM(final Connection conn, final Host host, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.startOnAsync(conn, host, false, true);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\", e);\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch HandleInvalid and VM is not in RUNNING state\");\n            } catch (final TimeoutException e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\", e);\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch BadAsyncResult and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            final String msg = \"Unable to start VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n            s_logger.warn(msg, e);\n            throw new CloudRuntimeException(msg);\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","date":"2018-04-16 15:07:16","endLine":5057,"groupId":"102935","id":4,"instanceNumber":2,"isCurCommit":0,"methodName":"startVM","params":"(finalConnectionconn@finalHosthost@finalVMvm@finalStringvmName)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/71/24659a6b5099a92b2474dfa2ca6d22c787f512.src","preCode":"    public void startVM(final Connection conn, final Host host, final VM vm, final String vmName) throws Exception {\n        Task task = null;\n        try {\n            task = vm.startOnAsync(conn, host, false, true);\n            try {\n                \r\n                waitForTask(conn, task, 1000, 10 * 60 * 1000);\n                checkForSuccess(conn, task);\n            } catch (final Types.HandleInvalid e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch HandleInvalid and VM is not in RUNNING state\");\n            } catch (final TimeoutException e) {\n                if (vm.getPowerState(conn) == VmPowerState.RUNNING) {\n                    s_logger.debug(\"VM \" + vmName + \" is in Running status\");\n                    task = null;\n                    return;\n                }\n                throw new CloudRuntimeException(\"Start VM \" + vmName + \" catch BadAsyncResult and VM is not in RUNNING state\");\n            }\n        } catch (final XenAPIException e) {\n            final String msg = \"Unable to start VM(\" + vmName + \") on host(\" + _host.getUuid() + \") due to \" + e.toString();\n            s_logger.warn(msg, e);\n            throw new CloudRuntimeException(msg);\n        } finally {\n            if (task != null) {\n                try {\n                    task.destroy(conn);\n                } catch (final Exception e1) {\n                    s_logger.debug(\"unable to destroy task(\" + task.toString() + \") on host(\" + _host.getUuid() + \") due to \" + e1.toString());\n                }\n            }\n        }\n    }\n","realPath":"plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":5021,"status":"M"}],"commitId":"106a9a34e12843a76cd67d9ab13837f56d79785a","commitMessage":"@@@Update inconsistent debugging info in catch block (#2553)\n\n* Add stack traces information\n\n* update stack trace info\n\n* update stack trace to make them consistent\n\n* update stack traces\n\n* update stacktraces\n\n* update stacktraces for other similar situations\n\n* fix some other situations\n\n* enhance other situations\n","date":"2018-04-16 15:07:16","modifiedFileCount":"11","status":"M","submitter":"lzh3636"}]
